apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/WriteBatch.cs

```diff
@@ -58,16 +57,13 @@ public WriteBatch Create(DocumentReference documentReference, object documentDat
         {
             GaxPreconditions.CheckNotNull(documentReference, nameof(documentReference));
             GaxPreconditions.CheckNotNull(documentData, nameof(documentData));
-            var write = new Write
-            {
-                CurrentDocument = Precondition.MustNotExist.Proto,
-                Update = new Document
-                {
-                    Fields = { ValueSerializer.SerializeMap(documentData) },
-                    Name = documentReference.Path
-                }
-            };
-            AddUpdate(write);
+            var fields = ValueSerializer.SerializeMap(documentData);
+            var serverTimestamps = new List<FieldPath>();
+            var deletes = new List<FieldPath>();
+            FindSentinels(fields, FieldPath.Empty, serverTimestamps, deletes);
```

Comment:
Could make `FindSentinels` return a `(List<FieldPath>, List<FieldPath>)` tuple?
(Or even better, use `ImmutableList<FieldPath>`)

Reply:
I suspect we'll want to introduce a dependency on ValueTuple eventually, but I'd rather put that off for now. (Especially as I seem to remember there are some issues with ValueTuple dependencies due to a mistake in one version. I'd need to validate that though.)

Fortunately all of this is internal - we can do it later if we want.